### PR TITLE
Updated project, source, and tests to Swift 3.0

### PR DESCRIPTION
**Pull request by regynald**
Updated cartfile to use 3.0 versions for both Quick & Nimble.
Had to remove typealias Middleware since Swift 3.0 doesn't allow @escaping in type aliases

### CODE REVIEW

**Comment by regynald** [comment]
Escaping change is [here](https://github.com/apple/swift/pull/4388/files)

File: Cartfile.private [hunk 1]
```diff
@@ -1,3 +1,2 @@
-github "Quick/Quick"
-github "Quick/Nimble"
-
+github "Quick/Quick" "swift-3.0"
+github "Quick/Nimble" "master"
```
Comment (lsunsi) [inline Cartfile.private]: this is fine for develop, but for a official master release we should probably wait until proper swift 3 support is released on both os this packages

File: ReduxSwift.playground/Contents.swift [hunk 2]
```diff
@@ -7,7 +7,7 @@ struct AppState {
         let text: String
         var done: Bool
     }
-
+
```
Comment (lsunsi) [inline ReduxSwift.playground/Contents.swift:10]: trailing whitespace. please use swiftlint (config file is included in the project) so we can maintain a coherent code style in the project. ps: this also applied for other styling changes

File: Sources/Core.swift [hunk 3]
```diff
@@ -8,18 +8,22 @@
...
+    open var dispatch: ((ActionType) -> ActionType)!
 
-    public init(reducer: Reducer, state: State?, middlewares: [Middleware]) {
+    public init(reducer: @escaping Reducer, state: State?, middlewares: [(Store<State>, @escaping Dispatcher) -> Dispatcher]) {
         self.state = state ?? reducer(INIT(), nil)
+        /*self.dispatch = middlewares.reduce({action in
+            self.state = reducer(action, self.state)
+            return action
+        }) {$1(self, $0)}*/
+
```
Comment (lsunsi) [inline Sources/Core.swift]: I don't get this comment. Seems to me it could just be removed.

File: Sources/Subscriptions.swift [hunk 4]
```diff
@@ -6,7 +6,7 @@
 //  Copyright © 2016 lsunsi. All rights reserved.
 //
 
-public class Subscriptions<State, Key: Hashable> {
+open class Subscriptions<State, Key: Hashable> {
```
Comment (lsunsi) [inline Sources/Subscriptions.swift]: maybe this class should be public and not open. what do you think?
my reasoning about this is that this class (and member) aren't intended for subclassing

**Review by lsunsi — CHANGES_REQUESTED** [review]
General comments on code style and some questions. Also, please merge on develop and not on master, so I can wait on Quick/Nimble next releases and properly tag ReduxSwift releases

File: Sources/Subscriptions.swift (hunk 4 above)
Comment (regynald) [reply Sources/Subscriptions.swift]: Yeah, I'm just gonna make the subscription class and associated functions public.

**Review by lsunsi — APPROVED** [review]

**Pull request merged**